### ISSUE TITLE
bud fails when run as non-root - error copying layers and metadata

### ISSUE THREAD

**Issue by TomSweeneyRedHat** [projectatomic/buildah#882]
**Description**

buildah bud fails when run as non-root user, but completes as root

**Steps to reproduce the issue:**
1.  $ cat Dockerfile
FROM fedora:28
RUN dnf -y update
2.  buildah bud -t myimg .

**Describe the results you received:**
```
$ buildah bud --net=host -t myimg .
STEP 1: FROM fedora:28
STEP 2: RUN dnf -y update
Fedora 28 - x86_64 - Updates                    2.5 MB/s |  20 MB     00:08    
Fedora 28 - x86_64                              892 kB/s |  60 MB     01:09    
Last metadata expiration check: 0:00:14 ago on Tue Jul 24 18:55:08 2018.
Dependencies resolved.
================================================================================
 Package                     Arch   Version                       Repository
                                                                           Size
================================================================================
Upgrading:
 acl                         x86_64 2.2.53-1.fc28                 updates  80 k
 {removed lots of happy update output for brevity}                                             
Complete!
STEP 3: COMMIT containers-storage:[vfs@/home/tom/.local/share/containers/storage+/run/user/1000/run]localhost/myimg:latest
Getting image source signatures
Skipping fetch of repeat blob sha256:891e1e4ef82ad02a4ea1f030831f942d722c7694c4db64ca3239c8163b811c58
Copying blob sha256:a6f9a4c6f5ae68a3090ddb2e73728d2a06778ba5b8fd316c90aac5697af9cc2a
 198.86 MiB / 198.86 MiB [==================================================] 5s
Copying config sha256:f1aaa659f928caf201d375680de469b568c6e8593c4f76cc4de2b187591e2860
 1.56 KiB / 1.56 KiB [======================================================] 0s
Writing manifest to image destination
Storing signatures
error copying layers and metadata: Error committing the finished image: error adding layer with blob "sha256:a6f9a4c6f5ae68a3090ddb2e73728d2a06778ba5b8fd316c90aac5697af9cc2a": ApplyLayer exit status 1 stdout:  stderr: lchown /etc/tcsd.conf: invalid argument
ERRO[0237] exit status 1   
```

**Describe the results you expected:**
A clean build.

**Output of `buildah version`:**

```
# buildah version
Version:         1.2
Go Version:      go1.10.3
Image Spec:      1.0.0
Runtime Spec:    1.0.0
CNI Spec:        0.3.1
libcni Version:  v0.6.0
Git Commit:      be87762
Built:           Sun Jul 15 07:05:04 2018
OS/Arch:         linux/amd64

```

**Output of `cat /etc/*release`:**

```
# cat /etc/*release
Fedora release 28 (Twenty Eight)
NAME=Fedora
VERSION="28 (Server Edition)"
ID=fedora
VERSION_ID=28
PLATFORM_ID="platform:f28"
PRETTY_NAME="Fedora 28 (Server Edition)"
ANSI_COLOR="0;34"
CPE_NAME="cpe:/o:fedoraproject:fedora:28"
HOME_URL="https://fedoraproject.org/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=28
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=28
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Server Edition"
VARIANT_ID=server
Fedora release 28 (Twenty Eight)
Fedora release 28 (Twenty Eight)
```

**Output of `uname -a`:**

```
# uname -a
Linux localhost.localdomain 4.17.7-200.fc28.x86_64 #1 SMP Tue Jul 17 16:28:31 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

```

**Output of `cat /etc/containers/storage.conf`:**

```
# cat /etc/containers/storage.conf
# storage.conf is the configuration file for all tools
# that share the containers/storage libraries
# See man 5 containers-storage.conf for more information
# The "container storage" table contains all of the server options.
[storage]

# Default Storage Driver
driver = "overlay"

# Temporary storage location
runroot = "/var/run/containers/storage"

# Primary Read/Write location of container storage
graphroot = "/var/lib/containers/storage"

[storage.options]
# Storage options to be passed to underlying storage drivers

# AdditionalImageStores is used to pass paths to additional Read/Only image stores
# Must be comma separated list.
additionalimagestores = [
]

# Size is used to set a maximum size of the container image.  Only supported by
# certain container storage drivers.
size = ""

# OverrideKernelCheck tells the driver to ignore kernel checks based on kernel version
override_kernel_check = "true"

# Remap-UIDs/GIDs is the mapping from UIDs/GIDs as they should appear inside of
# a container, to UIDs/GIDs as they should appear outside of the container, and
# the length of the range of UIDs/GIDs.  Additional mapped sets can be listed
# and will be heeded by libraries, but there are limits to the number of
# mappings which the kernel will allow when you later attempt to run a
# container.
#
# remap-uids = 0:1668442479:65536
# remap-gids = 0:1668442479:65536

# Remap-User/Group is a name which can be used to look up one or more UID/GID
# ranges in the /etc/subuid or /etc/subgid file.  Mappings are set up starting
# with an in-container ID of 0 and the a host-level ID taken from the lowest
# range that matches the specified name, and using the length of that range.
# Additional ranges are then assigned, using the ranges which specify the
# lowest host-level IDs first, to the lowest not-yet-mapped container-level ID,
# until all of the entries have been used for maps.
#
# remap-user = "storage"
# remap-group = "storage"

[storage.options.thinpool]
# Storage Options for thinpool

# autoextend_percent determines the amount by which pool needs to be
# grown. This is specified in terms of % of pool size. So a value of 20 means
# that when threshold is hit, pool will be grown by 20% of existing
# pool size.
# autoextend_percent = "20"

# autoextend_threshold determines the pool extension threshold in terms
# of percentage of pool size. For example, if threshold is 60, that means when
# pool is 60% full, threshold has been hit.
# autoextend_threshold = "80"

# basesize specifies the size to use when creating the base device, which
# limits the size of images and containers.
# basesize = "10G"

# blocksize specifies a custom blocksize to use for the thin pool.
# blocksize="64k"

# directlvm_device specifies a custom block storage device to use for the
# thin pool. Required if you setup devicemapper
# directlvm_device = ""

# directlvm_device_force wipes device even if device already has a filesystem
# directlvm_device_force = "True"

# fs specifies the filesystem type to use for the base device.
# fs="xfs"

# log_level sets the log level of devicemapper.
# 0: LogLevelSuppress 0 (Default)
# 2: LogLevelFatal
# 3: LogLevelErr
# 4: LogLevelWarn
# 5: LogLevelNotice
# 6: LogLevelInfo
# 7: LogLevelDebug
# log_level = "7"

# min_free_space specifies the min free space percent in a thin pool require for
# new device creation to succeed. Valid values are from 0% - 99%.
# Value 0% disables
# min_free_space = "10%"

# mkfsarg specifies extra mkfs arguments to be used when creating the base
# device.
# mkfsarg = ""

# mountopt specifies extra mount options used when mounting the thin devices.
# mountopt = ""

# use_deferred_removal Marking device for deferred removal
# use_deferred_removal = "True"

# use_deferred_deletion Marking device for deferred deletion
# use_deferred_deletion = "True"

# xfs_nospace_max_retries specifies the maximum number of retries XFS should
# attempt to complete IO when ENOSPC (no space) error is returned by
# underlying storage device.
# xfs_nospace_max_retries = "0"

```

**Comment by giuseppe** [comment]
is your user present in `/etc/subuid` and `/etc/subgid`?  How many extra IDs are present?

**Comment by TomSweeneyRedHat** [comment]
@giuseppe I installed a fresh vm, did an `adduser tom` and then `passwd tom`so that the only one added.  That's also the extent of my user setup/admin that I did for the tom account.  Here's the entries in /etc/subuid and /etc/subgid.

```
# cat /etc/subgid
tom:100000:65536

# cat /etc/subuid
tom:100000:65536
```

**Comment by giuseppe** [comment]
@TomSweeneyRedHat could you check if https://github.com/projectatomic/buildah/pull/884 solve the issue you are seeing?

**Comment by TomSweeneyRedHat** [comment]
This appears to have been fixed with #882
